components/backup-plugins/internal/plugins/nats-channels.go

```diff
@@ -0,0 +1,93 @@
+package plugins
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	"github.com/heptio/velero/pkg/plugin/velero"
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	evclientset "github.com/knative/eventing/pkg/client/clientset/versioned"
+	messagingv1alpha1Client "github.com/knative/eventing/pkg/client/clientset/versioned/typed/messaging/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	serviceNameSuffix = "-kn-channel"
+	channelKind       = "NatssChannel"
+)
+
+// SetOwnerReference is a plugin for velero to set new value of UID in metadata.ownerReference field
+// based on new values of other restored objects
+type SetNatsChannelOwnerReference struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo return list of resource kinds which should be handled by this plugin
+func (p *SetNatsChannelOwnerReference) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"service"},
+		LabelSelector:     "messaging.knative.dev/role=natss-channel",
+	}, nil
+}
+
+// Execute contains main logic for plugin
+// nolint
+func (p *SetNatsChannelOwnerReference) Execute(input *velero.RestoreItemActionExecuteInput) (
+	*velero.RestoreItemActionExecuteOutput, error) {
+	metadata, err := meta.Accessor(input.Item)
+	if err != nil {
+		return nil, err
+	}
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		p.Log.Infof("ERROR: getting cluster config: %v", err)
```

Comment:
It would be more idiomatic to return the error with a context (e.g. `error.Wrap(err, “getting cluster config)`), whatever executes the method will log it at the right log level (_not_ info).